src/Descriptor/AssemblyScanner.cs

```diff
@@ -12,10 +12,13 @@ public class AssemblyScanner : IEnumerable<IDescriptor<IDescriptorContainer>>
         public AssemblyScanner(IEnumerable<Type> scannerTypes)
         {
             this.scannerTypes = scannerTypes;
+            SetActivator(t => Activator.CreateInstance(t) as IDescriptor<IDescriptorContainer>);
         }
 
         private readonly IEnumerable<Type> scannerTypes;
 
+        public Func<Type, IDescriptor<IDescriptorContainer>> DescriptorActivator { get; set; }
```

Comment:
I'm okay with both. We can document the relationship between the two. The current-state matches the library's preference to let the user decide which to call. Some prefer fluent calls and others want a series of one-liners.